test/abstractarray.jl

```diff
@@ -1561,3 +1561,24 @@ end
     r = Base.IdentityUnitRange(3:4)
     @test reshape(r, :) === reshape(r, (:,)) === r
 end
+
+@testset "strides for ReshapedArray" begin
+    # Type-based contiguous check
+    a = vec(reinterpret(reshape,Int16,reshape(view(reinterpret(Int32,randn(Float64,100)),2:11),5,:)))
+    @test only(only(code_typed(Base._checkcontiguous, Base.typesof(Bool, a))).first.code).val
```

Comment:
Can you generalize this a bit? Checking that the inferred code consists of a single statement seems OK (though a comment explaining that's what you're doing would be helpful), but the `Const`-check seems fragile.

Reply:
I add some randomness on repshaped size in [ea36beb](https://github.com/JuliaLang/julia/pull/44027/commits/ea36beb7e60ac038b2b61f540053e5c5a65b00ce)
But I don't think it's actually better.
I consider to wrapped a `Vector` randomly, but it seems a little bit too complex for it.

Reply:
Sorry, I was referring to the fact that you're accessing the `CodeInfo` object and checking that the first statement is `Core.Compiler.Const(true)`. That's a deep dive into Julia's interals and might not be readable by future array-code maintainers who may know the array code well but may not know much about Julia's internal code representation. Also, if that representation changes, this test will break.

Reply:
I'll move this test to `test/compile/inline.jl`. (Where we have `fully_eliminated` defined.)
```julia
f(a) = only(strides(a));
@test fully_eliminated(f, Tuple{typeof(a)})
```
The above test would be equivalent. (It only use public api, so won't be infuenced by internal change.)